ext/fetch/22_body.js

```diff
@@ -64,17 +64,21 @@
   }
 
   class InnerBody {
+    #contentLength = null;
+
     /**
      * @param {ReadableStream<Uint8Array> | { body: Uint8Array | string, consumed: boolean }} stream
      */
-    constructor(stream) {
+    constructor(stream, { contentLength } = {}) {
       /** @type {ReadableStream<Uint8Array> | { body: Uint8Array | string, consumed: boolean }} */
       this.streamOrStatic = stream ??
         { body: new Uint8Array(), consumed: false };
       /** @type {null | Uint8Array | string | Blob | FormData} */
       this.source = null;
       /** @type {null | number} */
       this.length = null;
+
+      this.#contentLength = contentLength;
```

Comment:
Not sure yet what `this.length` is for, maybe it can be used instead of `this.#contentLength`